sdk/communication/communication-common/src/credential/communicationAccessKeyCredentialPolicy.ts

```diff
@@ -37,7 +37,7 @@ export function createCommunicationAccessKeyCredentialPolicy(
 
       const url = new URL(request.url);
       const query = url.searchParams;
-      const urlPathAndQuery = query ? `${url.pathname}?${query}` : url.pathname;
+      const urlPathAndQuery = query?.toString() ? `${url.pathname}?${query}` : url.pathname;
```

Comment:
One possible improvement might be doing `query?.toString()` on a separate line to do the string conversion just once. The `${url.pathname}?${query}` would then get string already.

Reply:
implemented this